money-core/src/main/scala/com/comcast/money/metrics/MetricRegistryFactory.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2012-2015 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.comcast.money.metrics
+
+import com.typesafe.config.Config
+import akka.actor.{ ActorSystem }
+import com.codahale.metrics.{ MetricRegistry, JmxReporter }
+import org.slf4j.LoggerFactory
+import scala.util.{ Try, Success, Failure }
+
+/*
+ * Simple factory that tries to delegate to an implementation of this trait itself and announced via
+ * the configuration 'metricRegistryFactory' - requiring that this implementation has a default constructor.
+ *
+ * It is up to the custom metricRegistryFactory what kind of MetricRegistry is created. No action is
+ * performed on this MetricRegistry is performed (i.e. registering reporters) but used/passed back right away.
+ *
+ * If any error occurs (configuration is not set, class can't be loaded etc.) the default behavior
+ * is performed and a fresh MetricRegistry is created and registered with the JmxReporter.
+ *
+ * Note: This trait should be kept as simple as possible so that the resulting interface can also be implemented
+ * by a Java client custom factory.
+ */
+object MetricRegistryFactory {
+  private val logger = LoggerFactory.getLogger("com.comcast.money.metrics.MetricRegistryFactory")
+
+  def metricRegistry(config: Config): MetricRegistry = {
+    Try({
+      // Try to create an instance of the custom factory, configured via 'metricRegistryFactory'
+      lazy val realFactory = Class.forName(config.getString("metrics-registry.class-name"))
```

Comment:
actually  can this lazy val realFactory be at the MetricRegistryFactory level?

That way it gets built once the first time its accessed.